pkg/serializer/jsonstream/compressor.go

```diff
@@ -69,15 +66,23 @@ type compressor struct {
 	repacks             int    // numbers of time we had to pack this payload
 	maxUnzippedItemSize int
 	maxZippedItemSize   int
+	maxPayloadSize      int
+	maxUncompressedSize int
 }
 
 func newCompressor(input, output *bytes.Buffer, header, footer []byte) (*compressor, error) {
+	// the backend accepts payloads up to 3MB compressed / 50MB uncompressed but
+	// prefers small uncompressed payloads of ~4MB
+	maxPayloadSize := config.Datadog.GetInt("serializer_max_payload_size")
+	maxUncompressedSize := config.Datadog.GetInt("serializer_max_uncompressed_payload_size")
```

Comment:
we're probably not in a hot loop, but config `get`s are quite slow. If you have benchmarks confirming this doesn't slow anything down you can disregard.

Later, could be nice to  make these 2 config params arguments of `newCompressor`, would probably simplify the tests as well

Reply:
It's not a hot path, we create one compressor per payload.